library/core/src/ptr/const_ptr.rs

```diff
@@ -1796,37 +1796,71 @@ impl<T: ?Sized> PartialOrd for *const T {
 mod verify {
     use crate::kani;
 
-    #[kani::proof_for_contract(<*const i32>::offset)]
-    fn check_offset_slice_i32(){
-        let mut arr: [i32; 5] = kani::any();
-        let test_ptr: *const i32 = arr.as_ptr();
-        let offset: isize = kani::any();
-
-        unsafe{
-            let new_ptr = test_ptr.offset(offset);
-        }
-    }
-
-    #[kani::proof_for_contract(<*const i32>::add)]
-    fn check_add_slice_i32() {
-        let mut arr: [i32; 5] = kani::any();
-        let test_ptr: *const i32 = arr.as_ptr();
-        let count: usize = kani::any();
-        unsafe {
-            let new_ptr = test_ptr.add(count);
-        }
-    }
+    // Constant for array size used in all tests
+    const ARRAY_SIZE: usize = 5;
+
+    /// This macro generates verification harnesses for the `offset`, `add`, and `sub`
+    /// pointer operations for a specific type and function name.
+    /// - `$ty`: The type of the array (e.g., i32, u32, tuples).
+    /// - `$offset_fn`: The function name for the `offset` operation.
+    /// - `$add_fn`: The function name for the `add` operation.
+    /// - `$sub_fn`: The function name for the `sub` operation.
+    macro_rules! generate_pointer_harnesses {
+        ($ty:ty, $offset_fn:ident, $add_fn:ident, $sub_fn:ident) => {
+            // Generates a harness for the `offset` operation
+            #[kani::proof_for_contract(<*const $ty>::offset)]
+            fn $offset_fn() {
+                let arr: [$ty; ARRAY_SIZE] = kani::Arbitrary::any_array();
+                let test_ptr: *const $ty = arr.as_ptr();
+                let offset: isize = kani::any();
+                unsafe {
+                    let new_ptr = test_ptr.offset(offset);
```

Comment:
can we offset test_ptr the non-deterministically within the allocated object, before we call offset on it? For example:
```rust
let offset: size = kani::any_where(|x| *x <= mem::size_of::<$ty>())
test_ptr = unsafe { test_ptr.byte_add(offset) }
```

Reply:
Updated in latest commit